application-collabora-default/src/main/java/com/xwiki/collabora/internal/FileTokenManager.java

```diff
@@ -128,28 +128,38 @@ public boolean isInvalid(String fileId, DocumentReference userReference)
     }
 
     /**
-     * Checks if a token is associated with the given file id and user and if the user has rights on the file, extends
-     * time out of the token.
+     * Checks if a token is associated with the given file id and user and if the user has rights on the file.
      *
      * @param fileId id of the edited file
      * @param userReference {@link DocumentReference} user reference associated with the checked token
-     * @return {@code true} if the token has been extended, {@code false} otherwise
+     * @return {@code true} if the token can be extended, {@code false} otherwise
      */
     public boolean canExtendToken(String fileId, DocumentReference userReference)
     {
         String user = userReference != null ? this.referenceSerializer.serialize(userReference) : XWIKI_GUEST;
         FileToken fileToken = getExistingToken(user, fileId);
         if (fileToken != null) {
             updateAccessRights(fileToken);
-            if (fileToken.hasView() || fileToken.hasEdit()) {
-                fileToken.extendTokenTimeout(configurationProvider.get().getTokenTimeout());
-                return true;
-            }
+            return hasAccess(fileToken.toString());
```

Comment:
I would do the `updateAccessRights(fileToken)` inside `hasAccess` (which I am surprised wasn't there already) and use directly the `hasAccess` method, since I don't feel that `canExtendToken` is adding any more value. Or I am missing something. WDYT?

Reply:
The `getExistingToken` function from `FileTokenManager` is private, so, to be able to create the token String I'd need to make it public or create a new `hasAccess` function that has as paramters the `fileId` and `userReference` needed to find the token. I chose the latter, as I find that the rest handler should not have to generate the token.